### PR TITLE
CF-1299: change the way we include sample XML and JSON by using xsdxt:code

### PR DESCRIPTION
**Pull request by shintasmith**
This PR fixes a few issues:
* updated the way we include XML and JSON samples in our DocBook and RST documentation. Instead of using:
```
<programlisting language="xml">
```
which does not work with wadl2rst, we are now using:
```
<para>
        <emphasis role="bold">XML Sample</emphasis>
        <xsdxt:code href="file:/Users/shin4590/src/forks/standard-usage-schemas/target/docbkx/message_samples/customer_access_policy/xml/customerservice-access_policy-info-v1-response.xml"/>
</para>
```
Both DocBook and RST seem to include the sample files. Further formatting may still be needed.
* stripped off comments from the samples XML files and placed them in ```target/docbkx/message_samples``` directory, as part of the main build
* response code 200 does not have any title and description in the RST files (see #534)
* pip install of wadl2rst now uses virtualenv (to fix the issue of the doc Jenkins build not having the right permissions when running pip install -- as it attempted to install in system directories).

### CODE REVIEW

**Comment by shintasmith** [comment]
[feeds-devguide-20151030.pdf](https://github.com/rackerlabs/standard-usage-schemas/files/26411/feeds-devguide-20151030.pdf)

**Comment by shintasmith** [comment]
@usnavi, @stanzikratel, @meker12, @ChandraAddala, please review.

File: api-docs/dev-guide/api-operations/methods/get-get-autoscale-event-autoscale-events-tid-entries-id.rst [hunk 1]
```diff
@@ -14,73 +14,60 @@ This http request fetches one particular event whose ID is listed in the URI.
...
...m:entry xmlns:atom="http://www.w3.org/2005/Atom" xmlns:xsd="http://www.w3.org/2001/XMLSchema" xmlns="http://www.w3.org/2001/XMLSchema" > < atom:id > urn:uuid:e53d007a-fc23-11e1-975c-cfa6b29bb814 < /atom:id > < atom:category term="rgn:DFW"/ > < atom:category term="dc:DFW1"/ > < atom:category term="autoscale.autoscale.info"/ > < atom:category term="type:autoscale.autoscale.info"/ > < atom:title > autoscale < /atom:title > < atom:content type="application/xml" > < event xmlns="http://docs.rackspace.com/core/event" xmlns:sample="http://docs.rackspace.com/event/autoscale" id="e53d007a-fc23-11e1-975c-cfa6b29bb814" version="2" eventTime="2013-03-15T11:51:11Z" type="INFO" dataCenter="DFW1" region="DFW" > < sample:product serviceCode="Autoscale" version="1" scalingGroupId="6e8bc430-9c3a-11d9-9669-0800200c9a66" desiredCapacity="5" currentCapacity="3" message="Launching 2 servers"/ > < /event > < /atom:content > < atom:link href="https://ord.feeds.api.rackspacecloud.com/autoscale/events/entries/urn:uuid:e53d007a-fc23-11e1-975c-cfa6b29bb814" rel="self"/ > < atom:updated > 2013-03-01T19:42:35.507Z < /atom:updated > < atom:published > 2013-03-01T19:42:35.507 < /atom:published > < /atom:entry >``
```
Comment (usnavi) [inline api-docs/dev-guide/api-operations/methods/get-get-autoscale-event-autoscale-events-tid-entries-id.rst]: I see the comments are still in the XML example.  I assume this is intentional?

Comment (shintasmith) [inline api-docs/dev-guide/api-operations/methods/get-get-autoscale-event-autoscale-events-tid-entries-id.rst]: Good catch. I forgot to re-generate the RST. Will push another commit also to fix the 200 response coe.

**Comment by shintasmith** [comment]
@usnavi, I pushed newly generated RST files that should not have the comments anymore.

**Comment by usnavi** [comment]
lgtm.  @ChandraAddala can merge once the checks pass

File: product_schema_def/xsl/productSchema-wadl.xsl [hunk 2]
```diff
@@ -136,6 +137,8 @@
                         </xsl:if>
                         <response status="200">
                             <representation mediaType="application/atom+xml"/>
+                            <outwadl:doc xmlns:outwadl="http://wadl.dev.java.net/2009/02" 
+                                         xmlns="http://docbook.org/ns/docbook" title="OK" xml:lang="EN">The request completed successfully</outwadl:doc>
```
Comment (meker12) [inline product_schema_def/xsl/productSchema-wadl.xsl]: In the DocBook output, the 200 response is still empty.   The product.wadl shows this code: 

````
<response status="200">
               <representation mediaType="application/atom+xml" title="OK" xml:lang="EN">The request completed successfully</representation>
            </response>
````
Missing closing tag after ```application/atom+xml```.   Missing wadl:doc tag.

I think you can fix this issue by removing the "xmlns:outwadl="http://wadl.dev.java.net/2009/02", unless you need it for something else.   Also, I wasn't sure why you have "outwadl:doc" .

What I realized is that the default spec defined is the java.net one.   If you use the default, then you can use "doc" tags to delimit doc elements.  However, for the DocBook transform, you want to override the default by using the wadl:doc tags with the DocBook wadl namespace. 

That's how the 400 response is defined:  
````
<response status="400">
                            <wadl:doc xmlns="http://docbook.org/ns/docbook" title="Bad Request" xml:lang="EN">The request is missing one or more elements, or the values of some 
                                elements are invalid.</wadl:doc>
                        </response>
````

**Comment by meker12** [comment]
So the 201 response and the 40x responses sourced from the  ProductSchema.xsl  all show up in the Docbook. 

They are coded like this:
```
 <!-- Okay -->
                    <response status="201" >
            			<wadl:doc title="Created">The request has been fulfilled. The entry has been created.</wadl:doc>
        			</response>
                    <!-- On Error -->
                    <response status="400">
                            <wadl:doc xmlns="http://docbook.org/ns/docbook" title="Bad Request" xml:lang="EN">The request is missing one or more elements, or the values of some 
                                elements are invalid.</wadl:doc>
                        </response>
                        <response status="401">
                            <wadl:doc xmlns="http://docbook.org/ns/docbook" title="Unauthorized" xml:lang="EN">Authentication failed, or the user does not have permissions for a requested operation.</wadl:doc>
                        </response>
                        <response status="409">
                            <wadl:doc xmlns="http://docbook.org/ns/docbook" title="The object already exists." xml:lang="EN">Duplicate entry ID sent 
                                in request. Fix entry and repost.
                            </wadl:doc>
                        </response>
                        <response status="500">
                            <wadl:doc xmlns="http://docbook.org/ns/docbook" title="Internal Server Error" xml:lang="EN">The server encountered an unexpected condition which prevented it from fulfilling the request.</wadl:doc>
                        </response>
                        <response status="503">
                            <wadl:doc xmlns="http://docbook.org/ns/docbook" title="Service Unavailable" xml:lang="EN">Service is not available. Try again later.</wadl:doc>
                    </response>

```

Can you try using one of these options for the 200 response code: 

```
<response status="201" >
            			<wadl:doc title="OK">The request succeeded.</wadl:doc>
   </response>
```
or this:

```
<response status="200">
                            <wadl:doc xmlns="http://docbook.org/ns/docbook" title="OK" xml:lang="EN">The request succeeded.</wadl:doc>
                    </response>
```

**Comment by shintasmith** [comment]
@meker12, my apologies.. I forgot to upload new PDFs and product.wadl yesterday after I fixed some of the issues. Here they are:
* product.wadl: [product.wadl.txt](https://github.com/rackerlabs/standard-usage-schemas/files/27113/product.wadl.txt)
* external guide: [feeds-devguide-20151103.pdf](https://github.com/rackerlabs/standard-usage-schemas/files/27114/feeds-devguide-20151103.pdf)

**Comment by shintasmith** [comment]
@meker12, I changed the ```<para>``` stuff as you requested. However, that doesn't seem to change the RST output. Let's handle that in a separate PR. Once you got a chance to review this PR and merge it, I will issue separate PRs for the minor changes: #543

**Comment by shintasmith** [comment]
@meker12, ok for me to merge this PR?

**Comment by ChandraAddala** [comment]
Looks Good.

**Pull request merged**